numpyro/infer/autoguide.py

```diff
@@ -146,6 +146,10 @@ def _setup_prototype(self, *args, **kwargs):
                 model_args=args,
                 model_kwargs=kwargs,
             )
+        # We apply a fixed seed just in case postprocess_fn requires
+        # a random key to generate subsample indices. It does not matter
+        # because we only collect deterministic sites.
```

Comment:
hmm what do you mean by 'does not matter'? what if the deterministic site depends on the subsample statement?

Reply:
oh gotcha this is in `_setup_prototype`